monitorcontroller/monitorcontroller.go

```diff
@@ -206,9 +205,14 @@ func (monitor *MonitorController) addNodeMonitoring(nodeMonitoring aostypes.Node
 	latestMessage := &monitor.offlineMessages[len(monitor.offlineMessages)-1]
 
 	// add node monitoring data
-	nodeDataInd := slices.IndexFunc(latestMessage.Nodes,
-		func(item cloudprotocol.NodeMonitoringData) bool { return item.NodeID == nodeMonitoring.NodeID },
-	)
+	nodeDataInd := -1
```

Comment:
nodeDataFound := false

	for i, nodeData := range latestMessage.Nodes {
		if nodeData.NodeID == nodeMonitoring.NodeID {
			latestMessage.Nodes[nodeDataInd].Items = append(
				latestMessage.Nodes[nodeDataInd].Items, nodeMonitoring.NodeData)

			nodeDataFound = true
			break
		}
	}

	if !nodeDataFound {
		latestMessage.Nodes = append(latestMessage.Nodes,
			cloudprotocol.NodeMonitoringData{
				NodeID: nodeMonitoring.NodeID, Items: []aostypes.MonitoringData{nodeMonitoring.NodeData},
			},
		)
	}